web/modules/custom/server_general/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,196 @@
+
+    return $build;
+  }
+
+  /**
+   * Build the header.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \IntlException
+   */
+  protected function buildHeader(NodeInterface $entity): array {
+    $elements = [];
+
+    $elements[] = $this->buildConditionalPageTitle($entity);
+
+    // Show the node type as a label.
+    $node_type = NodeType::load($entity->bundle());
+    $elements[] = $this->buildLabelsFromText([$node_type->label()]);
+
+    $current_user = \Drupal::currentUser();
+    //$logged_in = \Drupal::currentUser()->isAuthenticated();
+
+    if ($current_user->isAuthenticated()) {
+      $account = \Drupal\user\Entity\User::load($current_user->id());; // pass your uid
+      $name = $account->get('name')->value;
+
+      $parameters = [
+          'entity_type_id' => $entity->getEntityTypeId(),
+          'group' => $entity->id(),
+          'og_membership_type' => OgMembershipInterface::TYPE_DEFAULT,
+        ];
+
+      $url = Url::fromRoute('og.subscribe', $parameters)->toString();
+      $link = \Drupal\Core\Link::fromTextAndUrl('here', \Drupal\Core\Url::fromUri('internal:'.$url))->toString();
+
+      $elements[] = [
+        '#type' => 'markup',
+        '#markup' => t('Hi @user, click @link if you would like to subscribe to this group called @groupname', array('@user'=> $name, '@groupname' => $entity->label(), '@link' => $link)),
+      ];
+
+     /* $link['title'] = $this->t("Hi @user, click here if you would like to subscribe to this group called @groupname", ['@user'=> $name, '@groupname' => $entity->label()]);
+      $link['class'] = ['subscribe', 'request'];
+      $link['url'] = $url;
+
+      $elements[] = [
+        '#type' => 'link',
+        '#title' => $link['title'],
+        '#url' => $link['url'],
+      ];*/
+
+    }
+
+
+    // Date.
+    $timestamp = $this->getFieldOrCreatedTimestamp($entity, 'field_publish_date');
+    $element = IntlDate::formatPattern($timestamp, 'long');
+    // Make text bigger.
+    $elements[] = $this->wrapTextDecorations($element, FALSE, FALSE, 'lg');
+
+    $elements = $this->wrapContainerVerticalSpacing($elements);
+    return $this->wrapContainerNarrow($elements);
+  }
+
+  /**
+   * Build the Main content and the sidebar.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array
+   *
+   * @throws \IntlException
+   */
+  protected function buildMainAndSidebar(NodeInterface $entity): array {
+    $main_elements = [];
+    $sidebar_elements = [];
+    $social_share_elements = [];
+
+    $medias = $entity->get('field_group_image')->referencedEntities();
+    $main_elements[] = $this->buildEntities($medias);
+
+    
+    // Get the body text, wrap it with `prose` so it's styled.
+    $main_elements[] = $this->buildProcessedText($entity, 'body');
+
+    $main_elements[] = $entity->get('og_group')->view();
```

Comment:
Why are you adding this?

Reply:
I have used default body field of content type so I added to show that texts. 
And there is default field og_group for subscribe link in group content type so that I added that field to show subscribe link in full view.

Reply:
But that's not the requirement. Instead, you should add the custom text with the link.

Reply:
yes, right but I want to check how to add other fields so I added. I will solve other issues tomorrow morning.

Reply:
I see, but It's better if you stick to the requirements

Reply:
Correct :)